plugins/glustershd/glustershd.go

```diff
@@ -62,7 +60,13 @@ func (shd *Glustershd) Args() []string {
 
 // SocketFile returns path to the socket file used for IPC.
 func (shd *Glustershd) SocketFile() string {
-	return ""
+	if shd.socketfilepath != "" {
+		return shd.socketfilepath
+	}
+	glusterdSockDir := config.GetString("rundir")
```

Comment:
Use this value directly. A variable is needed if the value is used or set in more than one place.